Document NO_PROXY configuration and matching rules (#37802)
DataDog/documentation: content/en/agent/configuration/proxy.md

DD_LOGS_CONFIG_FORCE_USE_HTTP=true
```

## `NO_PROXY` Accepted Values

By default, `no_proxy`/`NO_PROXY` must match endpoints exactly for Agent HTTP(S) requests (except requests performed by Agent integrations). It is recommended to enable `no_proxy_nonexact_match` to make the Agent match `NO_PROXY` values with the same rules used for Agent integrations.

{{< tabs >}}
{{% tab "Configuration file" %}}
```yaml
no_proxy_nonexact_match: true
```
{{% /tab %}}
{{% tab "Environment variable" %}}
```bash
DD_NO_PROXY_NONEXACT_MATCH=true
```
{{% /tab %}}
{{< /tabs >}}

The following rules apply to Agent integrations (and the whole Agent when `no_proxy_nonexact_match` is enabled):
* A domain name matches that name and all subdomains, for example:
  - `datadoghq.com` matches `app.agent.datadoghq.com`, `www.datadoghq.com`, `datadoghq.com`, but **not** `www.notdatadoghq.com`

>   - `datadoghq` matches `frontend.datadoghq`, `backend.datadoghq`, but **not** `www.datadoghq.com` nor `www.datadoghq.eu`

Reviewer: Agree with this suggestion, please update - thank you!

DD_LOGS_CONFIG_FORCE_USE_HTTP=true
```

### Kubernetes

{{< tabs >}}
{{% tab "Helm" %}}

Add the following to your `values.yaml`:

```yaml
datadog:
  env:
    - name: DD_PROXY_HTTP
      value: "http://<PROXY_SERVICE>.<PROXY_NAMESPACE>.svc.cluster.local:<PROXY_PORT>"
    - name: DD_PROXY_HTTPS
      value: "http://<PROXY_SERVICE>.<PROXY_NAMESPACE>.svc.cluster.local:<PROXY_PORT>"
    - name: DD_PROXY_NO_PROXY
      value: "<HOST_TO_BYPASS_1> <HOST_TO_BYPASS_2>"
    - name: DD_NO_PROXY_NONEXACT_MATCH
      value: "true"
```

{{% /tab %}}
{{% tab "Operator" %}}

Add the following to your `DatadogAgent` CR:

```yaml
spec:
  global:
    env:
      - name: DD_PROXY_HTTP
        value: "http://<PROXY_SERVICE>.<PROXY_NAMESPACE>.svc.cluster.local:<PROXY_PORT>"
      - name: DD_PROXY_HTTPS
        value: "http://<PROXY_SERVICE>.<PROXY_NAMESPACE>.svc.cluster.local:<PROXY_PORT>"
      - name: DD_PROXY_NO_PROXY
        value: "<HOST_TO_BYPASS_1> <HOST_TO_BYPASS_2>"
      - name: DD_NO_PROXY_NONEXACT_MATCH
        value: "true"
```

{{% /tab %}}
{{< /tabs >}}

## `NO_PROXY` accepted values

> By default, `no_proxy`/`NO_PROXY` must match endpoints exactly for Agent HTTP(S) requests (except requests performed by Agent integrations). It is recommended to enable `no_proxy_nonexact_match` to make the Agent match `NO_PROXY` values with the same rules used for Agent integrations.

Reviewer: ```suggestion
By default, `no_proxy`/`NO_PROXY` must match endpoints exactly for Agent HTTP(S) requests (except requests performed by Agent integrations). Datadog recommends enabling `no_proxy_nonexact_match` to make the Agent match `NO_PROXY` values with the same rules used for Agent integrations.
```
